tests/README.md

```diff
@@ -0,0 +1,475 @@
+[`~/consensus-specs/tests/core/pyspec/eth2spec/test/phase0/sanity/test_blocks.py`](https://github.com/ethereum/consensus-specs/blob/dev/tests/core/pyspec/eth2spec/test/phase0/sanity/test_blocks.py)).
+
+```python
+@with_all_phases
+@spec_state_test
+def test_prev_slot_block_transition(spec, state):
+    spec.process_slots(state, state.slot + 1)
+    block = build_empty_block(spec, state, slot=state.slot)
+```
+
+Build an empty block for the current slot.
+
+```python
+    proposer_index = spec.get_beacon_proposer_index(state)
+```
+
+Get the identity of the current proposer, the one for *this* slot.
+
+```python
+    spec.process_slots(state, state.slot + 1)
+```
+
+Transition to the new slot, which naturally has a different proposer.
+
+```python
+    yield 'pre', state
+    expect_assertion_error(lambda: transition_unsigned_block(spec, state, block))
+```
+
+Specify that the function `transition_unsigned_block` will cause an assertion error.
+You can see this function in 
+[`~/consensus-specs/tests/core/pyspec/eth2spec/test/helpers/block.py`](https://github.com/ethereum/consensus-specs/blob/dev/tests/core/pyspec/eth2spec/test/helpers/block.py),
+and one of the tests is that the block must be for this slot:
+> ```python
+> assert state.slot == block.slot
+>  ```
+
+Because we use [lambda notation](https://www.w3schools.com/python/python_lambda.asp), the test
+does not call `transition_unsigned_block` here. Instead, this is a function parameter that can
+be called later.
+
+```python
+    block.state_root = state.hash_tree_root()
+```
+
+Set the block's state root to the current state hash tree root, which identifies this block as
+belonging to this slot (even though it was created for the previous slot). 
+
+```python    
+    signed_block = sign_block(spec, state, block, proposer_index=proposer_index)
+```
+
+Notice that `proposer_index` is the variable we set earlier, *before* we advanced
+the slot with `spec.process_slots(state, state.slot + 1)`. It is not the proposer 
+for hte current state.
+
+```python
+    yield 'blocks', [signed_block]
+    yield 'post', None   # No post state, signifying it errors out
+```
+
+This is the way we specify that a test is designed to fail - failed tests have no post state,
+because the processing mechanism errors out before creating it.
+
+
+## Attestation Tests
+
+The [beacon chain](https://ethereum.org/en/eth2/beacon-chain/) doesn't provide any direct value. It does
+not execute EVM programs or store user data. It exists to provide a trusted source of
+information about the latest verified block hash of the [shard blockchains](https://ethereum.org/en/eth2/shard-chains/)
+which do provide storage, and possibly execution, services.
+
+For every slot a validator is randomly selected as the proposer. Currently the proposer proposes the hash
+for the current head of the beacon chain (the previous block). When shards are added, the proposer will also
+propose a hash for the head of the assigned shard.
```

Comment:
```suggestion
```

Reply:
I would just cut out a lot of the sharding stuff. There are actually separate proposrs that build the shard chains and it's probably not worth getting in the nuance here.